openedx/frontend-app-course-authoring
# feat: include usage locations in delete modal

## Description

This PR enhances the details in the files and videos delete confirmation. The new details clarify that deleting content can impact the rendering of the course if used in advanced settings, units, etc. Additional added a list of collapsible sections for the files/videos that are used in units so that users are aware that the files/videos that they are trying to delete might mess up content. This change impacts Authors.

Before
<img width="524" alt="Screenshot 2024-04-05 at 4 06 46 PM" src="https://github.com/openedx/frontend-app-course-authoring/assets/42981026/6ba9d928-dfe3-4111-98e8-0db2c3fa3c6c">

After
<img width="524" alt="Screenshot 2024-04-05 at 4 04 27 PM" src="https://github.com/openedx/frontend-app-course-authoring/assets/42981026/ee7f20f2-9351-4e7d-8bc3-5b5349ad19c4">

## Supporting information

JIRA Ticket: [TNL-10957](https://2u-internal.atlassian.net/browse/TNL-10957)

## Testing instructions

1. Navigate to the Files page
2. Select a file to delete that is not used in a course
3. Should open delete confirmation with the file name is the title
4. Should not show the list of files that are used in the course
5. Click "Cancel"
6. Select a file to delete that is used in acourse
7. Should open delete confirmation with the file name is the title
8. Should show the list of files that are used in the course
9. Click "Cancel"
10. Select multiple files
11. Should number of selected files in the title
12. Should show files that are used in the course listed
13. Should not show files that are not used in the course
